src/linker/Linker.Dataflow/XmlFlowAnnotationSource.cs

```diff
@@ -0,0 +1,595 @@
+			if (fieldAnnotation != DynamicallyAccessedMemberTypes.None)
+				_fields[field] = fieldAnnotation;
+		}
+
+		void ProcessFieldName (TypeDefinition type, string name, XPathNodeIterator iterator)
+		{
+			if (!type.HasFields)
+				return;
+
+			foreach (FieldDefinition field in type.Fields) {
+				if (field.Name == name) {
+					ArrayBuilder<Attribute> attributes = ProcessAttributes (iterator.Current.SelectChildren ("attribute", string.Empty));
+					DynamicallyAccessedMemberTypes fieldAnnotation = GetMemberTypesForDynamicallyAccessedMemberAttribute (attributes, _context, _xmlDocumentLocation);
+					if (fieldAnnotation != DynamicallyAccessedMemberTypes.None)
+						_fields[field] = fieldAnnotation;
+				}
+			}
+		}
+
+		static FieldDefinition GetField (TypeDefinition type, string signature)
+		{
+			if (!type.HasFields)
+				return null;
+
+			foreach (FieldDefinition field in type.Fields)
+				if (signature == field.FieldType.FullName + " " + field.Name)
+					return field;
+
+			return null;
+		}
+
+		void ProcessMethodSignature (TypeDefinition type, string signature, XPathNodeIterator iterator)
+		{
+			MethodDefinition method = GetMethod (type, signature);
+			if (method == null) {
+				_context.LogMessage (MessageContainer.CreateWarningMessage ($"Could not find method '{signature}' in type '{type.FullName}' specified in '{_xmlDocumentLocation}'", 2009));
+				return;
+			}
+			ProcessMethodChildren (type, method, iterator);
+		}
+
+		void ProcessMethodChildren (TypeDefinition type, MethodDefinition method, XPathNodeIterator iterator)
+		{
+			ArrayBuilder<Attribute> attributes = ProcessAttributes (iterator.Current.SelectChildren ("attribute", string.Empty));
+			ArrayBuilder<(string, ArrayBuilder<Attribute>)> parameterAnnotations = ProcessParameters (type,
+				method, iterator.Current.SelectChildren ("parameter", string.Empty));
+			ArrayBuilder<ArrayBuilder<Attribute>> returnParameterAnnotations = ProcessReturnParameters (type,
+				method, iterator.Current.SelectChildren ("return", string.Empty));
+
+			var parameterAnnotation = new ArrayBuilder<(string ParamName, DynamicallyAccessedMemberTypes Annotation)> ();
+			DynamicallyAccessedMemberTypes returnAnnotation = 0;
+
+			if (parameterAnnotations.Count > 0) {
+				foreach (var parameter in parameterAnnotations.ToArray ()) {
+					DynamicallyAccessedMemberTypes paramAnnotation = GetMemberTypesForDynamicallyAccessedMemberAttribute (parameter.Item2, _context, _xmlDocumentLocation);
+					if (paramAnnotation != 0)
+						parameterAnnotation.Add ((parameter.Item1, paramAnnotation));
+				}
+			}
+
+			if (returnParameterAnnotations.Count == 1) {
+				foreach (var returnparameter in returnParameterAnnotations.ToArray ()) {
+					DynamicallyAccessedMemberTypes returnparamAnnotation = GetMemberTypesForDynamicallyAccessedMemberAttribute (returnparameter, _context, _xmlDocumentLocation);
+					if (returnparamAnnotation != 0)
+						returnAnnotation = returnparamAnnotation;
```

Comment:
This on the other hand picks the last occurrence - maybe we should stick to the "first one wins" here as well.

Reply:
I'm not allowing to have more than one return. `returnParameterAnnotations` parses all the return parameters but I only try to parse if the return parameters count is 1. 
Not sure if I need to fail on these kinds of cases. If I fail people will need to fix it to continue, if I warn the return parameter will not be taken into account and we will continue, but I think that's the behavior of most of the members in the XML.